### DISCUSSION TITLE
Very confused about iOS 14 ad tracking

### DISCUSSION THREAD

**Discussion by ugrdursun** [invertase/react-native-firebase#5489] [General]
Hello, first of all, thanks for this beautiful library.

I am so confused about this warning: 

<img width="513" alt="Ekran Resmi 2021-07-07 15 48 01" src="https://user-images.githubusercontent.com/23141977/124762157-43e3d000-df3b-11eb-9d72-822983356c41.png">

My app works with RN 0.61.5 now and well, and I use firebase packages of these versions below:

`    "@react-native-firebase/admob": "^6.7.1",
    "@react-native-firebase/analytics": "^6.7.1",
    "@react-native-firebase/app": "^6.7.1",
    "@react-native-firebase/auth": "^6.7.1",
    "@react-native-firebase/database": "^6.7.1",
    "@react-native-firebase/dynamic-links": "^6.7.1",
    "@react-native-firebase/messaging": "^6.7.1",
    "@react-native-firebase/storage": "^6.7.1",`

So what should I do exactly for adapting my app to iOS 14 rules according to my versions ? 

Google says `Prerequisites: Google Mobile Ads SDK 7.64.0 or higher`, which version rnfirebase version includes that ? And how may i know what is my current version ?

If package update needed, which version do i need to upgrade and how will arrange the Ad tracking permission ?

Thanks

**Comment by mikehardy** [comment]
Hi there! Glad you like the library

Those versions, they are antique :-) - you're missing heaps of bugfixes.

If you use admob you can't come all the way to current but you should move to 11.5.x at least (last versions with admob)

In general though, this is not a question about the module though - it won't result in a change here, in other words. I think this might work better on stackoverflow as a question

**Reply by ugrdursun** [reply]
> Hi there! Glad you like the library
> 
> Those versions, they are antique :-) - you're missing heaps of bugfixes.
> 
> If you use admob you can't come all the way to current but you should move to 11.5.x at least (last versions with admob)
> 
> In general though, this is not a question about the module though - it won't result in a change here, in other words. I think this might work better on stackoverflow as a question


Is 11.5.x includes Google Mobile Ads SDK 7.64.0 or higher ?

Edit: I upgraded all packages to 11.5.x, and during pod update it said:

`Installing Google-Mobile-Ads-SDK 7.69.0 (was 7.67.0)`


Sooo it was already higher than 7.64.0. So receiving this warning is all about ATT permission ? And how does apple knows i dont use personalized adds even if user denied it :)
